src/System.IO.Compression/tests/ZipArchive/zip_ReadTests.cs

```diff
@@ -185,5 +184,16 @@ public class zip_ReadTests : ZipFileTestBase
 
             s.Dispose();
         }
+
+        [Fact]
+        public static async Task Crc32Test()
+        {
+            ZipArchive archive1 = new ZipArchive(await StreamHelpers.CreateTempCopyStream(zfile("normal.zip")), ZipArchiveMode.Read);
+            ZipArchive archive2 = new ZipArchive(await StreamHelpers.CreateTempCopyStream(zfile("normal.zip")), ZipArchiveMode.Read);
+            ZipArchive archive3 = new ZipArchive(await StreamHelpers.CreateTempCopyStream(zfile("small.zip")), ZipArchiveMode.Read);
+
+            Assert.True(archive1.Entries[0].Crc32 == archive2.Entries[0].Crc32);
+            Assert.True(archive1.Entries[0].Crc32 != archive3.Entries[0].Crc32);
```

Comment:
Can we validate that it's actually the correct value, and not just that two different inputs produce two different values?

Also, what about other ctors where we should be validating that Crc32 starts at 0 and then presumably that it changes?

Reply:
+1

Probably the best/easiest way to get the "correct" Crc32 values against which to compare our Crc32 would be to open up the test zips in something like 7zip and put those values in a const in the FileData file that contains all of the other hardcoded expected values.

Reply:
The CRC Values are already in the FileData

Reply:
@stephentoub I extended my tests. I just got the problem now, that the crc does not update when i insert a new entry in the archiv. 
It will update when the archive get's reopened